x-pack/docs/en/security/securing-communications/security-basic-setup.asciidoc

```diff
@@ -0,0 +1,277 @@
+[[security-basic-setup]]
+=== Set up basic security for the Elastic Stack
+++++
+<titleabbrev>Set up basic security</titleabbrev>
+++++
+
+After adding password protection in the <<security-minimal-setup,minimal security configuration>>, you'll need to configure Transport Layer Security
+(TLS). The transport layer handles all internal communication between nodes in
+your cluster.
+
+The transport layer relies on mutual TLS for both encryption and
+authentication. Applying TLS ensures that a malicious node cannot join the
+cluster and exchange data with other nodes. While implementing username and
+password authentication at the HTTP layer is useful for securing a local
+cluster, there is no internode security without configuring TLS.
+
+Configuring TLS is the basic security setup to prevent unauthorized nodes from
+accessing to your cluster.
+
+[[basic-setup-prerequisites]]
+==== Prerequisites
+
+Complete the steps in <<security-minimal-setup,Minimal security for the Elastic Stack>> to enable {es} security features on every node in your cluster. You can
+then encrypt communications between your nodes with TLS.
+
+NOTE: You only need to create passwords for the built-in users one time for the
+entire cluster.
+
+[[generate-certificates]]
+==== Generate the certificate authority
+
+You can add as many nodes as you want in a cluster but they must be able to communicate with each other. The communication between nodes in a cluster is handled by the transport module. To secure your cluster, you must ensure that the internode communications are encrypted and verified which is achieved with mutual TLS.
+
+In a secured cluster, {es} nodes use certificates to identify
+themselves when communicating with other nodes.
+
+The cluster must validate the authenticity of these certificates. The
+recommended approach is to trust a specific certificate authority (CA). When
+nodes are added to your cluster they must use a certificate signed by the same
+CA.
+
+For the transport layer, we recommend using a separate, dedicated CA instead
+of an existing, possibly shared CA so that node membership is tightly controlled. Use the `elasticsearch-certutil` tool to
+generate a CA for your cluster.
+
+NOTE: To generate certificates for a multi-node cluster, jump to
+<<multi-node-certs,Generate certificates for a multi-node cluster>> and use
+that method of generating a CA and certificates.
```

Comment:
Per the earlier question raised by @jkakavas, this seems a little weird. 
If you don't have a multi-node cluster, then don't have internode communication, so there's nothing to set up.

So, we have 2 options listed here one of which is easier, but we essentially tell people not to use it (because they have multiple nodes).
I think that means we need to pick a recommendation and tell people to follow that path, and then maybe offer the alternative as a separate "if you wish ..."

The getting started guide recommended the single certificate option because it's as simple as possible. You have 1 file that you copy to each node and any node that has that file can join the cluster, and nodes that don't have it can't. There's no need to keep a CA key around and generate new certificates when you add new nodes, you just copy the same PKCS#12 file each time, and it works correctly with dynamic IP addresses/hostnames.

However, it gives up some security properties (full verification mode) in order to do that. Not everyone is convinced of the tradeoff. So we have the 2nd option for "multiple certificates". But that second option has pitfalls. 
- To add new nodes to the cluster you need to generate new certificates (or have pre-emptively generated multi-san/wildcard certs), which means keeping track of the CA key (and keeping it safe, or it defeats the purpose).
- Doing the above is very hard if you want the cluster to scale dynamically. The orchestrator needs to have the CA key and generate a new certificate each time.
- It's hard to make it work if you have dynamic IP addresses. You need to use DNS SANs instead, but then you also need to configure each ES node to publish its transport address using the DNS name, not the IP.
- It's impossible if you have dynamic IPs and dynamic hostnames.

Given those issue, the only scenario that works every time is single certificate, which is my recommendation. I'm OK with us having the "cert per node" option documented somewhere, but I don't think we should be encouraging people to use it unless they understand the pitfalls (and really, how can we expect them to?)

Reply:
Thanks for the explanation here @tvernum. I'm going to extract the section for full verification mode with one certificate per node, and focus only on the scenario that generates a single certificate. We can add a page for full verification mode in a follow-up PR.